azext_iot/_params.py

```diff
@@ -232,6 +232,13 @@ def load_arguments(self, _):
             arg_type=get_three_state_flag(),
             help="Flag indicating edge enablement.",
         )
+        context.argument(
+            "connection_string",
+            options_list=["--connection-string", "--cs"],
+            help="Target connection string. This bypasses the IoT Hub registry and generates the SAS token directly"
+            " from the supplied symmetric key without further validation. All other parameters aside from duration"
+            " will be ignored."
```

Comment:
Sorry - was writing a comment and realized I was just confused by this - we're ignoring "command parameters" not parts of the connection string :)

Just a thought.

Reply:
In here I did mean command parameters. Fixed.

Another thought: in the connection string, can the user have extra key/values? Currently the code will ignore them.

ex: `HostName=hubhostname;SharedAccessKeyName=iothubowner;SharedAccessKey=blah;Extra=extra_value`

Reply:
Technically - sure, they can add anything they want to the string, but if it's not part of the connection string "spec" then it will and should be ignored, and of course not used to generate the SAS token.